runtime/compiler/env/VMJ9.cpp

```diff
@@ -1213,13 +1213,68 @@ TR_J9VMBase::getObjectClassFromKnownObjectIndex(TR::Compilation *comp, TR::Known
    return clazz;
    }
 
+TR_OpaqueClassBlock *
+TR_J9VMBase::getObjectClassFromKnownObjectIndex(TR::Compilation *comp,
+                                                TR::KnownObjectTable::Index idx,
+                                                bool isJavaLangClass)
+   {
+   TR::VMAccessCriticalSection vpKnownObjectCriticalSection(comp);
+
+   TR::KnownObjectTable *knot = comp->getKnownObjectTable();
+   if (!knot)
+      return NULL;
+
+   TR_OpaqueClassBlock *clazz = getObjectClass(knot->getPointer(idx));
+   TR_OpaqueClassBlock *jlClass = getClassClassPointer(clazz);
+   if (isJavaLangClass)
+      {
+      TR_ASSERT(clazz == jlClass,
```

Comment:
I was thinking that this assert should be moved to the caller, so that it triggers at the server side if needed, not at the client side.
So, what I am proposing is for this method to return `isJavaLangClass` based on `clazz==jlClass` equality and then the caller, in this case the VP code executed by the server check that the value of returned `isJavaLangClass` matches its expectation.